.github/workflows/deploy.yml

```diff
@@ -0,0 +1,65 @@
+name: deploy
+
+"on":
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+  workflow_run:
+    workflows: [ "test" ]
```

Comment:
Now, the question here is whether this will make the workflow trigger whenever the test workflow is executed, including when it's executed on PRs? Hmm. If that's so, then we need to do something else.

Reply:
We need to only trigger deployment when a change was merged, *and* the "test" workflow executed successfully. Could you figure out how to make that happen?

Reply:
Perhaps we need to use this instead: https://docs.github.com/en/actions/learn-github-actions/reusing-workflows

In essence, we need the "test" workflow run on pull requests, and when a change is merged we need to first run the "test" workflow, and then the "deploy" workflow.